### PR TITLE
Terraform to spin up X number of VPCs with VPN Gateways.

### PR DESCRIPTION
**Pull request by acormier-maia**
Initial working of Terraform to scale test VPN.

- Create N number of VPCs, each with VPN Gateway
- EC2 VM autostarts with iperf3 server on each interface.
- Create the VPN Connections

In the future we could probably support scaling up number of EC2 instances to remove the limit for number of interfaces on a single one.  We could potentially use routing tricks in the VPC to support more than one VPN per VPC.

ENG-928

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -0,0 +1,34 @@
+locals {
```
Comment (abilash-maiatech) [inline main.tf:1]: do we need to check in the lock file ?

File: modules/iperf_ec2/main.tf [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+  ip_protocol       = "tcp"
+  cidr_ipv4         = "0.0.0.0/0"
+}
+
+# Egress rule for all (to install iperf)
+resource "aws_vpc_security_group_egress_rule" "iperf_all" {
+  for_each = { for idx in range(var.number_of_subnets) : tostring(idx) => idx }
+
+  security_group_id = aws_security_group.iperf[each.key].id
+  ip_protocol       = "-1"
```
Comment (abilash-maiatech) [inline modules/iperf_ec2/main.tf:76]: any reason egress rule takes everything ?

**Review by abilash-maiatech — APPROVED** [review]

File: modules/iperf_ec2/main.tf (hunk 2 above)
Comment (acormier-maia) [reply modules/iperf_ec2/main.tf:76]: Basically so I can access internet from the VMs to install iperf3

File: main.tf (hunk 1 above)
Comment (acormier-maia) [reply main.tf:1]: Yes, lockfiles are checked in.

**Pull request merged**